src/TileSource.js

```diff
@@ -169,6 +169,7 @@ class TileSource {
       return true;
     } catch (error) {
       return error.code === 'EEXIST';
+      throw error;
```

Comment:
The `return` above will never make the runtime reach the `throw` statement. Maybe `if (error.code === 'EEXIST') return true;` is intended?